// python -m grpc_tools.protoc -Iprotos/ --python_out=pythonserver/ --grpc_python_out=pythonserver/ protos/server.proto

syntax = "proto3";

package server;

service SocketGuide {

    rpc GetSocketFamilyList(SocketTree) returns (stream SocketFamily) {}

    rpc GetSocketTypeList(SocketFamily) returns (stream SocketType) {}

    rpc GetSocketProtocolList(SocketType) returns (stream SocketProtocol) {}

}

message SocketTree {
    string choice = 1;
}

message SocketFamily {
    string name = 1;
    int32 value = 2;
}

message SocketType {
    string name = 1;
    int32 value = 2;
}

message SocketProtocol {
    string name = 1;
    int32 value = 2;
}